### PR TITLE
added metrics for read and write

### PR DESCRIPTION
**Pull request by hervenicol**
Towards https://github.com/giantswarm/giantswarm/issues/23523

Add metrics to the dependency-based chart.
It creates 2 additional services, with the label and annotation needed so that they are discovered and monitored by our MC setup.

Tested on gauss: works, we have our metrics. It successfully scrapes each pod of the service.
image

However, it also discovers the other port, and obviously fails scraping it.
image
This means we have some faulty targets in Prometheus. Not very clean. :disappointed: 

Do you have some suggestions to make it better?

### CODE REVIEW

**Comment by QuentinBisson** [comment]
How about using a service monitor instead? Or a PodMonitor if that works better?

**Comment by hervenicol** [comment]
> How about using a service monitor instead? Or a PodMonitor if that works better?

That would be the best, as it is the recommended and supported upstream solution.
But I understood it will take some time before prometheus-operator runs on all workload clusters, so we need a short-term solution in the meantime.

File: helm/loki/values.yaml [hunk 1]
```diff
@@ -29,6 +29,10 @@ multiTenantAuth:
 
 imagePullSecrets: []
 
+giantswarm:
+  labelmonitoring:
```
Comment (TheoBrigitte) [inline helm/loki/values.yaml:33]: Why `labelmonitoring` instead of `monitoring` ?

**Review by TheoBrigitte — COMMENTED** [review]
I do not understand how this produce a target with port `9095`, maybe something is wrong in the service selector ?

File: helm/loki/values.yaml (hunk 1 above)
Comment (hervenicol) [reply helm/loki/values.yaml:33]: As opposed to the "normal" monitoring discovery which would be with servicemonitors rather than labels.

**Comment by hervenicol** [comment]
> I do not understand how this produce a target with port `9095`, maybe something is wrong in the service selector ?

It's the GRPC port of Loki service. It probably means the "monitoring-port" selector does not work. :thinking:

**Comment by hervenicol** [comment]
With port number (`3100`) instead of alias name (`http-metrics`) it works better!
There's no more failing scraping targets related to Loki now.

File: helm/loki/values.yaml (hunk 1 above)
Comment (TheoBrigitte) [reply helm/loki/values.yaml:33]: Ok make sense

**Comment by TheoBrigitte** [comment]
> With port number (`3100`) instead of alias name (`http-metrics`) it works better! There's no more failing scraping targets related to Loki now.

Nice :+1:

**Review by TheoBrigitte — APPROVED** [review]

**Pull request merged**